### DISCUSSION TITLE
Help on pastash conf file

### DISCUSSION THREAD

**Discussion by spady7** [sipcapture/homer#652] [Q&A] [answered]
I am trying to make working this configuration, but seams something is wrong.
Only "tags" CDR apply. Not AC_SYSLOG and also output not working based on IF statement.
Can you point me in the right direction?

```
input {
  udp {
    host => 0.0.0.0
    port => 9999
    tags => 'AC_SYSLOG'
type => 'sbc_logs'
  }
  udp {
    host => 0.0.0.0
    port => 514
    tags => 'CDR'
type => 'cdr'
  }
}

filter {
  if [type] == "sbc_logs"{
  app_audiocodes{
debug => true
}
  }
 
 
  if [type] == "cdr" {
  compute_field {
    field => origin
    value => "AUDIOCODES"
}
  json_fields {}
}
}

output {
 if [type] == "sbc_logs" {
    stdout {}
      hep {
        host => "heplify-server"
        port => 9060
        hep_id => 100
      }
   
  }
  else if [type] == "cdr" {
    stdout {}
    loki {
      host => "qryn"
      port => 3100
      path => "/loki/api/v1/push"
    }
  }
}
```

**Comment by lmangani** [comment]
Hey @spady7 could you describe your status a little better, showing examples of what works and what doesn't?

**Reply by spady7** [reply]
Hi @lmangani sure.
I've and SBC (audiocodes) that sends both syslog messages and cdr messages. Then I've a docker server where is running your homer 10 stack (https://github.com/sipcapture/homer-docker) in which I've added pastash container to handle audiocodes syslog and i want to add also handling of CDR packets.
The idea was to intercept (INPUT) in different port those 2 type of messages, apply a tag and then output based on tag (or type) to a different endpoint (loki or heplify-server).
More or less this is idea.
First, is possibile what i thought? Then, why is not working?
If you need incoming messages i can provide them.
Regards

**Reply by lmangani** [reply]
Thanks for the clarification. While this is possible, I would not advice mixing different flows. Due to the very basic IF/ELSE capabilities in the pastash plan, It's much more efficient to have separate pastash instances handling individual recipes rather than mixing them.

With this said - the recipe should work to some extent although sending meaningful hep logs requires a correlation_id to be present or you simply will never see this data. What does the stdout{} print when the hep output is reached?

**Reply by spady7** [reply]
OK, got it.
I changed, for test, splitting configuration in 2 separate CONF files (as below):

```
input {
  udp {
    host => 0.0.0.0
    port => 514
    tags => 'CDR'
  }
}

filter {
  compute_field {
    field => origin
    value => "AUDIOCODES"
	}
  json_fields {}
}


output {
    stdout {}
	file {
		path => "/tmp/cdr_#{now:YYYY-MM}.log"
		serializer => json_logstash
	}
#    loki {
#      host => "qryn"
#      port => 3100
#      path => "/loki/api/v1/push"
#    }
  }



```
and

```
input {
  udp {
    host => 0.0.0.0
    port => 9999
    tags => 'AC_SYSLOG'
	type => 100
  }
}

filter {
  app_audiocodes{
	debug => true
	}
  }

output {
    stdout {}
	file {
		path => "/tmp/syslog_#{now:YYYY-MM}.log"
		serializer => json_logstash
	}
  }


```

Now as result i have 2 files (like output configured) but with same content. Why? Should be one with CDR content and one with SYSLOG content. Am I wrong?

**Reply by lmangani** [reply]
Nothing in the code or example would be able to do that. You must have other instances running or something with bad configs.

**Reply by spady7** [reply]
ohhh sorry. You mean, in my case I am running it in container, using 2 separate container?

**Reply by lmangani** [reply, marked as answer]
I can't tell without knowing how it's built, but you should be on the right path forward. Next time, best to open an issue ;)
